smithy-crt/build.gradle.kts

```diff
@@ -9,7 +9,7 @@ repositories {
 }
 
 plugins {
-    kotlin("jvm") version "1.4.20"
+    kotlin("jvm") version "1.7.10"
```

Comment:
This fixes the build error. This ancient version of Kotlin was failing when running JDK 17.

I'm specifying the latest stable version of Kotlin.

I don't **THINK** (???) it matters, since this is just our codegen? And we ultimately compile the generated code to target JDK 1.8? Right?